flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1232,21 +1232,31 @@ def run_to_convergence(self, desired_datasets, max_iterations=4):
 
     @given(
```

Comment:
You could probably do something with either `flatmap` or more plausibly `composite`, which is documented later in that page: sample an initial dataset. then sample a next state, and for each additional field in the new type sample values from relevant strategy.